cmd/monaco/cmdutils/cmdutils.go

```diff
@@ -40,19 +40,19 @@ func SilenceUsageCommand() func(cmd *cobra.Command, args []string) {
 // CreateDTClient is driven by data given through a manifest.EnvironmentDefinition to create an appropriate client.Client.
 //
 // In case when flag dryRun is true this factory returns the client.DummyClient.
-func CreateDTClient(env manifest.EnvironmentDefinition, dryRun bool) (client.Client, error) {
+func CreateDTClient(env manifest.EnvironmentDefinition, dryRun bool, opts ...func(dynatraceClient *client.DynatraceClient)) (client.Client, error) {
```

Comment:
why not put dryRun also as optional fucntion parameter

Reply:
it's evaluated in the factory function, if it's set we are passing back ka DummyClient. So it doesn't make sense to pass it as a client option

Reply:
For now `CreateDTClient` decides based on information what client implementation to hand back.
`DryRun` is one of the parameters this decision is based on. It has nothing to do with the internals of a specific client implementation. So your point still doesn't make sense to me, sorry.

Reply:
I would even say it's the opposite of what you say what would be more consistent: passing the limiter parameters as required (non optional) parameter.